src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -2033,11 +2033,11 @@ void MacroAssembler::fat_nop() {
   if (UseAddressNop) {
     addr_nop_5();
   } else {
-    emit_int8((int8_t)0x26); // es:
-    emit_int8((int8_t)0x2e); // cs:
-    emit_int8((int8_t)0x64); // fs:
-    emit_int8((int8_t)0x65); // gs:
-    emit_int8((int8_t)0x90);
+    emit_int8((uint8_t)0x26); // es:
+    emit_int8((uint8_t)0x2e); // cs:
+    emit_int8((uint8_t)0x64); // fs:
+    emit_int8((uint8_t)0x65); // gs:
+    emit_int8((uint8_t)0x90);
   }
```

Comment:
Maybe there us a better way to do this with a template `emit_int8()` function, but maybe not now.

Reply:
I fixed these casts as an alternative to the emit_int8 templates, so left them in this patch because they're more correct than what was there.  The emit_intXX functions need special treatment that I don't know how to fix right now.